### PR TITLE
created class for data sharing

### PR DESCRIPTION
**Pull request by sadiabbasi**

### CODE REVIEW

File: src/sdk/util/dataSharingRequest.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import { PublicKey } from '@wharfkit/antelope';
+import { VCWithTypeType, VerifiableCredentialOptions, VerifiableCredentialWithType } from './ssi/vc';
+import { Issuer } from '@tonomy/did-jwt-vc';
+
+export type DataSharingRequestPayload = {
+    username: string;
+    origin: string;
```
Comment (theblockstalk) [inline src/sdk/util/dataSharingRequest.ts:7]: only the username should be shared here. No other data needs to be shared here.

File: src/sdk/util/dataSharingRequest.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+     */
+    constructor(vc: DataSharingRequest | VCWithTypeType<DataSharingRequestPayload>) {
+        super(vc);
+        this.decodedPayload.publicKey = PublicKey.from(super.getPayload().publicKey);
+    }
+
+    /**
+     * Alternative constructor that returns type DataSharingRequest
+     */
+    static async createRequest(
```
Comment (theblockstalk) [inline src/sdk/util/dataSharingRequest.ts:26]: should be `signRequest()` to stay consistent with other requests

**Review by theblockstalk — COMMENTED** [review]
I think you need to perhaps also change the response from login so that it doesn't have username. Or this can be in a future task

https://github.com/Tonomy-Foundation/Tonomy-ID-SDK/blob/master/src/sdk/controllers/userApps.ts#L114

**Comment by sadiabbasi** [comment]
> I think you need to perhaps also change the response from login so that it doesn't have username. Or this can be in a future task
> 
> https://github.com/Tonomy-Foundation/Tonomy-ID-SDK/blob/master/src/sdk/controllers/userApps.ts#L114

we can do it in next task when we update login with tonomy

**Review by theblockstalk — APPROVED** [review]

**Pull request merged**